notebooks/03-query/09-functions-case.ipynb

```diff
@@ -0,0 +1,499 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  REGEXP_EXTRACT(\"loaded_image\",'[^/\\\\&\\?]+\\.\\w{3,4}(?=([\\?&].*$|$))') AS \"filename\",\n",
+    "  SUM(CASE WHEN (\"event_type\" = 'GoodLoad') THEN 1 ELSE 0 END) AS \"impressions\",\n",
+    "  SUM(CASE WHEN (\"event_subtype\" IS NULL AND \"event_type\" = 'LayerClear') THEN 1 ELSE 0 END) AS \"conversions\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T0/PT4H')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9a995573-8991-4f89-a36b-deabcf70cd86",
+   "metadata": {},
+   "source": [
+    "Run the following cell to return a new dimension that takes the percent cleared and turns it into a funnel stage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38778ef0-93dd-4635-b5e4-f770ec16b38e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  CASE\n",
+    "    WHEN \"event_subtype\" <= 24 THEN '1'\n",
+    "    WHEN \"event_subtype\" BETWEEN 25 AND 49 THEN '2'\n",
+    "    WHEN \"event_subtype\" BETWEEN 50 AND 74 THEN '3'\n",
+    "    WHEN \"event_subtype\" BETWEEN 75 AND 99 THEN '4' END\n",
+    "  AS \"funnelStage\",\n",
+    "  COUNT(*) AS \"Count\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE \"event_type\" = 'PercentClear'\n",
+    "AND TIME_IN_INTERVAL(__time,'2019-08-25T0/PT4H')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "917dd59a-0f1f-41b0-90ed-6b990ea696d0",
+   "metadata": {},
+   "source": [
+    "Because the data has been recorded in order, we can combine this `CASE` with `MAX` to determine how far down the funnel each session went for each image within a specific time window."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ac699351-b8c3-4d93-8be3-64e227c5f496",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  session,\n",
+    "  REGEXP_EXTRACT(\"loaded_image\",'[^/\\\\&\\?]+\\.\\w{3,4}(?=([\\?&].*$|$))') AS \"filename\",\n",
+    "  MAX(\n",
+    "    CASE\n",
+    "      WHEN \"event_subtype\" <= 24 THEN 1\n",
+    "      WHEN \"event_subtype\" BETWEEN 25 AND 49 THEN 2\n",
+    "      WHEN \"event_subtype\" BETWEEN 50 AND 74 THEN 3\n",
+    "      WHEN \"event_subtype\" BETWEEN 75 AND 99 THEN 4 END)\n",
+    "  AS \"funnelStage\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE \"event_type\" = 'PercentClear'\n",
+    "AND TIME_IN_INTERVAL(__time,'2019-08-25T0/PT4H')\n",
+    "GROUP BY 1, 2\n",
+    "LIMIT 10\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3fee335b-2bbb-467f-bebc-f8626397b258",
+   "metadata": {},
+   "source": [
+    "## Using CASE at ingestion time\n",
```

Comment:
Running the notebook, I almost missed that this section was about using CASE at _ingestion_ as opposed to querying. Could you expand on example use cases or why someone would want to do this at ingestion?

It would also be helpful to give more details what the following table holds -- a plain English version of the SQL statement if you will. :)

Reply:
I reworded a few bits and expanded... thanks!